server/src/admin/admin-routes.ts

```
@@ -472,7 +488,22 @@ export const adminAddUsersToWorkshop = async function addUsersToWorkshop(request
     for (const email of emailList) {
       const user = await checkForUsersWithUsernameOrEmail(email, email);
       if (user.length > 0) {
-        await updateUser({ email }, { workshopId });
+        for (const eachUser of user) {
+          const { userId } = eachUser;
+          const existingRelationship = await checkWorkshopHasUser(userId, workshopId);
+          if (existingRelationship.length == 0) {
+            await createUsersWorkshops({ userId: userId, workshopId: workshopId });
+            const newRelationship = await checkWorkshopHasUser(userId, workshopId);
+            if (newRelationship.length !== 1) {
+              reply.status(500).send({ error: "Error adding user to workshop", invalidEmails: email });
+              return;
+            }
+          } else {
+            //TODO: add test
+            reply.status(500).send({ error: "Duplicated user-workshop relationship", invalidEmails: email });
+            return;
```

do you need to throw an error if the user is already in the workshop? i don't think you need to, you could just continue because they already are a part of the workshop? @asivath

---

also if you added test remove the TODO, if you haven't yet, then write the test